CMakeLists.txt

```diff
@@ -31,6 +31,12 @@ ENDIF()
 ADD_DEFINITIONS(-DDEVICE_API="${DEVICE_API}")
 message(STATUS "Setting DEVICE_API to ${DEVICE_API}")
 
+IF (NOT DEFINED GIT_COMMIT)
+    execute_process(COMMAND sh -c "git describe --match=NeVeRmAtCh --always --dirty | tr -d '\n'" OUTPUT_VARIABLE GIT_COMMIT)
+ENDIF()
```

Comment:
I would just prefer for not calling git directly here as this software could as well be distributed as a tarball later on.

What is usually done here is setting the default to 'unknown' and allowing the user to customize via -D, then we can change the OE recipe to feed the variable as part of the build process.

Reply:
that works for me. i'd copied the aktualizr approach but they had an error case for when git wasn't available. I'll do -D since we are doing that everywhere else.

Reply:
rebased and force pushed